File: Day6/Decimal To Binary/C++/DecimalToBinary.cpp
#include <iostream>
#include <math.h>
#include <cmath>

using namespace std;

int main()
{
    long n;
    cin >> n;

    long ans = 0;
    long count = 0;

    while(n != 0)
    {
        long bit = n & 1;
       
       ans += (bit * ceil(pow(10,count))); 
        
        count++;        
        n = n >> 1;
    }

    cout << ans;
    return 0;
}

File: Day6/Decimal To Binary/Python/DecimalToBinary.py
def PositiveConversion(n):
    count = 0
    ans = 0

    while n != 0:
        bit = n & 1

        ans += (bit * 10**count)

        n = n >> 1
        count += 1
    
    return ans


n = int(input())

ans = PositiveConversion(n)

print(ans)    